repository/src/main/java/org/alfresco/repo/content/transform/TransformerDebug.java

```diff
@@ -775,77 +711,79 @@ private String getReference(boolean firstLevelOnly, boolean overrideFirstLevel,
             }
             else
             {
-                sb.append(spaces(13-sb.length()+lengthOfFirstId)); // Try to pad to level 7
+                sb.append(spaces(REFERENCE_SIZE-sb.length()+lengthOfFirstId)); // Try to pad to level 7
             }
         }
         return sb.toString();
     }
 
-    public String getFileName(NodeRef sourceNodeRef, boolean firstLevel, long sourceSize)
-    {
-        return getFileNameOrNodeRef(sourceNodeRef, firstLevel, sourceSize, true);
-    }
-
-    private String getNodeRef(NodeRef sourceNodeRef, boolean firstLevel, long sourceSize)
-    {
-        return getFileNameOrNodeRef(sourceNodeRef, firstLevel, sourceSize, false);
-    }
-    
-    private String getFileNameOrNodeRef(NodeRef sourceNodeRef, boolean firstLevel, long sourceSize, boolean getName)
+    public String getFilename(NodeRef sourceNodeRef, boolean firstLevel)
     {
-        String result = getName ? null : "";
+        String result = null;
         if (sourceNodeRef != null)
         {
             try
             {
-                result = getName
-                        ? (String)nodeService.getProperty(sourceNodeRef, ContentModel.PROP_NAME)
-                        : sourceNodeRef.toString()+" ";
+                result = (String)nodeService.getProperty(sourceNodeRef, ContentModel.PROP_NAME);
             }
             catch (RuntimeException e)
             {
-                // ignore (normally InvalidNodeRefException) but we should ignore other RuntimeExceptions too
+                // ignore (InvalidNodeRefException/MalformedNodeRefException) but we should ignore other RuntimeExceptions too
             }
         }
-        if (result == null)
+        if (result == null && !firstLevel)
         {
-            if (!firstLevel)
-            {
-                result = getName ? "<<TemporaryFile>>" : "";
-            }
-            else if (sourceSize < 0)
-            {
-                // fileName = "<<AnyFile>>"; commented out as it does not add to debug readability
-            }
+            result = "<<TemporaryFile>>";
         }
         return result;
     }
 
+    private String getNodeRef(NodeRef sourceNodeRef)
+    {
+        return sourceNodeRef == null ? "" : sourceNodeRef.toString();
+    }
+
     protected String getSourceAndTargetExt(String sourceMimetype, String targetMimetype)
     {
         String sourceExt = getMimetypeExt(sourceMimetype);
         String targetExt = getMimetypeExt(targetMimetype);
-        targetExt = AsynchronousExtractor.getExtension(targetMimetype, sourceExt, targetExt);
+        targetExt = replaceWithMetadataExtensionIfEmbedOrExtract(targetMimetype, sourceExt, targetExt);
         return sourceExt + targetExt + spaces(1+4-targetExt.length());
     }
 

```

Comment:
Similar code in alfresco-transform-model - getting closer to joining the two together.